pennylane_ionq/api_client.py

```diff
@@ -112,12 +112,16 @@ class APIClient:
         api_key (str): IonQ cloud platform API key
     """
 
-    USER_AGENT = "pennylane-ionq-api-client/0.1"
-    HOSTNAME = "api.ionq.co/v0.1"
+    USER_AGENT = "pennylane-ionq-api-client/0.3"
+    HOSTNAME = "api.ionq.co/v0.3"
     BASE_URL = "https://{}".format(HOSTNAME)
 
     def __init__(self, **kwargs):
-        self.AUTHENTICATION_TOKEN = os.getenv("IONQ_API_KEY") or kwargs.get("api_key", None)
+        self.AUTHENTICATION_TOKEN = (
+            kwargs.get("api_key", None)
+            or os.getenv("PENNYLANE_IONQ_API_KEY")
+            or os.getenv("IONQ_API_KEY")
+        )
```

Comment:
yeah, this reordering is intentional - also PENNYLANE_IONQ_API_KEY is added because sometimes we have different clients interact in different ways in tests, so this just specifies the pennylane client